### PR TITLE
Fix stair-case solver in 2D Cartesian

### PR DESCRIPTION
**Pull request by RemiLehe**
The stair-case solver does not work properly for the `Ey` component in 2D Cartesian. This PR fixes this, both for the vacuum Maxwell solver and the macroscopic Maxwell solver.

### CODE REVIEW

File: Examples/Modules/embedded_boundary_cube/analysis_fields_2d.py [hunk 1]
```diff
@@ -48,14 +48,17 @@
...
 rel_tol_err = 1e-3
 
 # Compute relative l^2 error on By
 By_sim = data['By'].to_ndarray()
 rel_err_y = np.sqrt(np.sum(np.square(By_sim - By_th)) / np.sum(np.square(By_th)))
 assert (rel_err_y < rel_tol_err)
 
+# Compute relative l^2 error on Ey
+Ey_sim = data['Ey'].to_ndarray()
+rel_err_y = np.sqrt(np.sum(np.square(Ey_sim/c - By_th)) / np.sum(np.square(By_th)))
```
Comment (lgiacome) [inline Examples/Modules/embedded_boundary_cube/analysis_fields_2d.py:60]: Shouldn't it be Ey_th here?

Comment (RemiLehe) [reply Examples/Modules/embedded_boundary_cube/analysis_fields_2d.py:60]: Yes, but in this case (i.e. with the initial conditions that I provided in the input script), `Ey_th` should simply be proportional to `By_th` (i.e. `Ey_th = c*By_th`), so I did not bother recalculating it.

Comment (lgiacome) [reply Examples/Modules/embedded_boundary_cube/analysis_fields_2d.py:60]: Ah perfect! Thanks for the clarification

**Review by ax3l — APPROVED** [review]
Thanks, LGTM!

Any additional branch needed for 1D?

**Pull request merged**